### ISSUE TITLE
Look into multiple dispatch helpers for cleaning up GUI event code

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#2313]
## Enhancement request:

[singledispatchmethod]: https://docs.python.org/3/library/functools.html#functools.singledispatchmethod

**TL;DR:** Consider one of the other dispatch tools to clean up GUI event code

### What should be added/changed?

Consider one of the following to clean up our event code:
1. [`functools.singledispatchmethod`][singledispatchmethod]
2. The 3rd party [`multipledispatch`](https://github.com/mrocklin/multipledispatch/)
3. Beartype's [`plum`](https://github.com/beartype/plum)

### Why wait til 3.0+?
[singledispatch]: https://docs.python.org/3/library/functools.html#functools.singledispatch
There might be unexpected problems which would delay 3.0.

The status of the following are unclear:
1. `Union` types and arguments
   * [`functools.singledispatch`][singledispatchmethod] only got `Union` support in 3.11
   * Where does that leave [`singledispatchmethod`][singledispatchmethod]?
3. It looks like [there are issues with inheritance as of a month ago](https://stackoverflow.com/questions/68190543/how-to-use-singledispatchmethod-with-inheritance-class)

### What would it help with?

Reduce verbosity in cases like this:
https://github.com/pythonarcade/arcade/blob/05b476f5bbe3e02a2a7352d67e5daa980e7f489b/arcade/gui/widgets/text.py#L432-L477

**Comment by cspotcode** [comment]
Bear in mind, every if statement is going to become an annotated method
declaration, so total lines of code may increase.

**Comment by eruvanos** [comment]
Hi,

from my point of view, we do not have to enforce a decision
for such topic without experimenting and testing it in praxis.

You can just create a UIWidget, which passes the events to the singledispatch function.
This could even be an opt-in Mix-in.

__Side note:__

I personally try to stick to the current arcade "statement":

> Arcade is an easy-to-learn Python library for creating 2D video games. It is ideal for beginning programmers, or programmers who want to create 2D games without learning a complex framework.

My conclusion is, it should be easy to understand for beginner, or have to be optional.
( I am aware, that the GUI comes with its own complexity, but it is the easiest solution I found so far 🙈)

**Comment by einarf** [comment]
Let's put this one to sleep for now. It's something we can revisit in the distant future. We can't break 3.0 now.
